Doctrine/Orm/OrderFilter.php

```diff
@@ -0,0 +1,102 @@
+{
+    /**
+     * @var array List of properties by witch the collection can or cannot be ordered.
+     */
+    private $sourceList = [];
+
+    /**
+     * @var bool If true $this#sourcelist is used as a whitelist. Otherwise is used as a blacklist.
+     */
+    private $type = true;
+
+    /**
+     * @param array   $properties List of property names on which the filter will be enabled/disabled.
+     * @param boolean $type       If true properties are passed as a whitelist and as a blacklist otherwise.
+     */
+    public function __construct(array $properties = [], $type = true)
+    {
+        $this->sourceList = $properties;
+        $this->type       = $type;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Order collection by properties. The order of the ordered properties is the same as the order specified in the
+     * query.
+     */
+    public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder, Request $request)
+    {
+        $this->applyFilter($resource, $queryBuilder, $request->query->get('order'));
+    }
+
+    /**
+     * Create query to order the collection by the properties passed.
+     * For each property passed, if the resource does not have such property or if the order value is different from
+     * `asc` or `desc` (case insensitive), the property is ignored.
+     *
+     * @param ResourceInterface $resource
+     * @param QueryBuilder      $queryBuilder
+     * @param array             $values       Array of properties as key and order value as value.
+     *
+     * @return QueryBuilder
+     */
+    protected function applyFilter(ResourceInterface $resource, QueryBuilder $queryBuilder, array $values)
+    {
+        $metadata      = $queryBuilder->getEntityManager()->getClassMetadata($resource->getEntityClass());
+        $fieldNames    = $metadata->getFieldNames();
+        $filterList    = [];
+
+        foreach ($values as $property => $order) {
+            $order = strtoupper($order);
+
+            if (true === in_array($property, $fieldNames)                   // Check if the entity has the property
+                && $this->type === in_array($property, $this->sourceList)   // Check if property is enabled
+                && ('ASC' === $order || 'DESC' === $order)                  // Check if order value is valid
+            ) {
+                $queryBuilder->addOrderBy(sprintf('o.%s', $property), $order);
+            }
+        }
+
+        return $queryBuilder;
+    }
+
+    /**
+     * Gets the description of this filter for the given resource.
+     * Returns an array with the filter names as keys and array with the following data as values:
+     *   - type: the type of the filter
+     *   - strategy: the used strategy
+     * The description can contain additional data specific to a filter.
+     *
+     * @param ResourceInterface $resource
+     *
+     * @return array
+     */
+    public function getDescription(ResourceInterface $resource)
```

Comment:
I'm not sure of how this method should be implemented: how is it supposed to be used outside the class?